src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -79,6 +79,14 @@ internal static class ParameterHelpers
                     diagnostics.Add(ErrorCode.ERR_IllegalRefParam, refnessKeyword.GetLocation());
                 }
 
+                if (parameterSyntax.ExclamationToken.Kind() == SyntaxKind.ExclamationToken)
+                {
+                    if (owner.IsAbstract || owner.IsImplementableInterfaceMember() || owner.IsPartialDefinition() || owner.IsExtern)
+                    {
+                        diagnostics.Add(ErrorCode.ERR_MustNullCheckInImplementation, parameterSyntax.Location, parameterSyntax.ToString());
```

Comment:
>parameterSyntax.ToString() [](start = 111, length = 26)

This will be the entire parameter syntax including type and default value, etc. Consider using `parameterSyntax.Identifier.ValueText` instead, as in other diagnostics that include the parameter name.